storage/src/main/kotlin/maru/database/kv/KvDatabase.kt

```diff
@@ -60,6 +67,13 @@ class KvDatabase(
   override fun getSealedBeaconBlock(beaconBlockRoot: ByteArray): SealedBeaconBlock? =
     kvStoreAccessor.get(Schema.SealedBeaconBlockByBlockRoot, beaconBlockRoot).getOrNull()
 
+  override fun getSealedBeaconBlock(blockNumber: ULong): SealedBeaconBlock? {
+    kvStoreAccessor.get(Schema.BlockRootByBlockNumber, blockNumber).getOrNull()?.let { blockRoot ->
+      return getSealedBeaconBlock(blockRoot)
+    }
+    return null
```

Comment:
I don't think you need a separate return statement. I believe you should be able to pop it before `kvStoreAccessor.get(...`